### ISSUE TITLE
PyLance reports subscript will generate runtime exception within TYPE_CHECKING clause

### ISSUE THREAD

**Issue by cpnielsen** [microsoft/pylance-release#2728]
## Environment data

- VS Code plugin version: 2022.4.2
-  Language Server version: Pylance language server 2022.4.2 (pyright 90161194)
-  OS and version: macOS Monterey (12.2.1)
-  Python version: 3.7.12 (via `asdf`)

## Code Snippet / Additional information

```
import asyncio
from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    asyncTask = asyncio.Task[Any]
else:
    asyncTask = asyncio.Task
    
 <other code>
```

I do not have `pyright` installed (I am using `mypy`, `flake8` and `black` alongside pylance).

## Expected behavior

No errors shown in "Problems" 

## Actual behavior

I get the following error:

```plain
Subscript for class "Task" will generate runtime exception; enclose type annotation in quotes
```

Changing it to `asyncTask = asyncio.Task["Any"]` does not resolve the error.

**Comment by erictraut** [comment]
This behavior is intended. The `TYPE_CHECKING` symbol is designed to "lie" to type checker about what the Python interpreter will see at runtime. I recommend avoiding the use of `TYPE_CHECKING` if at all possible because it leads to differences between runtime and static analysis. This is an example.

The recommended approach here is to use a quoted form of the annotation, which doesn't generate a runtime exception when used in code. 

If you want to create a type alias for a quoted annotation, you can use the `TypeAlias` special form introduced in [PEP 613](https://peps.python.org/pep-0613/). 

```python
from typing_extensions import TypeAlias

asyncTask: TypeAlias = "asyncio.Task[Any]"
```

I'll note that the underlying problem (where `asyncio.Task` is not subscriptable at runtime) was fixed in Python 3.9, so once you move from Python 3.7 to a newer version, you can avoid these types of workarounds.

**Comment by cpnielsen** [comment]
@erictraut That makes sense, thank you for the thorough explanation 👍
